src/target/linux.rs

```diff
@@ -221,6 +306,192 @@ impl LinuxTarget {
 
         Ok(tasks)
     }
+
+    #[cfg(target_arch = "x86_64")]
+    pub fn set_hardware_breakpoint(
+        &mut self,
+        breakpoint: HardwareBreakpoint,
+    ) -> Result<usize, Box<dyn std::error::Error>> {
+        #[cfg(target_arch = "x86_64")]
+        {
+            let index = if let Some(empty) = self.find_empty_watchpoint() {
+                empty
+            } else {
+                return Err(Box::new(HardwareBreakpointError::NoEmptyWatchpoint));
+            };
+
+            let rw_bits: u64 = match breakpoint.typ {
+                HardwareBreakpointType::Execute => 0b00,
+                HardwareBreakpointType::Read => 0b11,
+                HardwareBreakpointType::ReadWrite => 0b11,
+                HardwareBreakpointType::Write => 0b01,
+            } << 16 + index * 4;
+
+            let size_bits: u64 = (breakpoint.size as u64) << (18 + index * 4);
```

Comment:
I'd suggest to move this to a function in `HardwareBreakpoint`, e.g.:

```rust
impl HardwareBreakpoint {
    fn size_bits(&self, index: usize) -> u64 {
        (self.size as u64) << (18 + index * 4)
    }
}
```
then it can be used locally as
```suggestion
            let size_bits = breakpoint.size_bits(index);
```